### ISSUE TITLE
phpunit.xml `SESSION_DRIVER` `cookie` does not work with testing `withSession`

### ISSUE THREAD

**Issue by kocoten1992** [laravel/framework#42298]
- Laravel Version: 9.11.0
- PHP Version: 8.1.5
- Database Driver & Version: -

### Description:
Accidentally discover when change `array` to `cookie` in phpunit.xml

```
<env name="SESSION_DRIVER" value="cookie"/>
```

### Steps To Reproduce:
1. change in phpunit.xml
```
<env name="SESSION_DRIVER" value="cookie"/>
```
2. run test with this:
```
$this->withSession(['this_will' => 'will_throw_exception']);
```

```
  Attempt to read property "cookies" on null

  at vendor/laravel/framework/src/Illuminate/Session/CookieSessionHandler.php:75
     71▕      * @return string|false
     72▕      */
     73▕     public function read($sessionId): string|false
     74▕     {
  ➜  75▕         $value = $this->request->cookies->get($sessionId) ?: '';
     76▕ 
     77▕         if (! is_null($decoded = json_decode($value, true)) && is_array($decoded) &&
     78▕             isset($decoded['expires']) && $this->currentTime() <= $decoded['expires']) {
     79▕             return $decoded['data'];

      +5 vendor frames
```

I'm guessing this is a papercut since [docs](https://laravel.com/docs/9.x/testing) doesn't mention about this.

**Comment by driesvints** [comment]
I don't think enabling a cookie session driver in unit tests makes sense.

**Comment by fulopattila122** [comment]
**tl;dr** to fix this, add/change `<env name="SESSION_DRIVER" value="array"/>` to/in your application's `phpunit.xml` file. (Alternatively in `.env.testing`)
